src/main.rs

```diff
@@ -1649,6 +1650,66 @@ fn cmd_regenerate_audit_as(
     Ok(())
 }
 
+fn cmd_renew(out: &Arc<dyn Out>, cfg: &Config, sub_args: &RenewArgs) -> Result<(), miette::Report> {
+    trace!("renewing wildcard audits");
+    let mut store = Store::acquire_offline(cfg)?;
+    do_cmd_renew(out, cfg, &mut store, sub_args);
+    store.commit()?;
+    Ok(())
+}
+
+fn do_cmd_renew(out: &Arc<dyn Out>, cfg: &Config, store: &mut Store, sub_args: &RenewArgs) {
+    assert!(sub_args.expiring ^ sub_args.crate_name.is_some());
+
+    let new_end_date = cfg.today + chrono::Months::new(12);
+
+    let mut renewing: WildcardAuditRenewal;
+
+    if let Some(name) = &sub_args.crate_name {
+        match WildcardAuditRenewal::single_crate(name, store) {
+            Some(renewal) => {
+                renewing = renewal;
+                if renewing.is_empty() {
+                    info!("no wildcard audits for {name} are eligible for renewal (all have `renew = false`)");
+                    return;
+                }
+            }
+            None => {
+                warn!("ran `renew {name}`, but there are no wildcard audits for the crate");
+                return;
+            }
+        }
+    } else {
+        // Find and update all expiring crates.
+        assert!(sub_args.expiring);
+        renewing = WildcardAuditRenewal::expiring(cfg, store);
+
+        if renewing.is_empty() {
+            info!("no wildcard audits that are eligible for renewal have expired or are expiring in the next {WILDCARD_AUDIT_EXPIRATION_STRING}");
+            return;
+        }
+    }
+
+    renewing.renew(new_end_date);
+
+    write!(
+            out,
+            "Updated wildcard audits for the following crates and user ids to expire on {new_end_date}:"
+        );
+    for (name, entries) in renewing.crates {
+        writeln!(
+            out,
+            "  {}: {:80}",
+            name,
+            // FormatShortList ends up sorting the strings by length and then
+            // lexicographically, which will serendipitously be the same as a numeric sort.
+            string_format::FormatShortList::new(
+                entries.iter().map(|a| a.0.user_id.to_string()).collect()
+            )
```

Comment:
It's a bit weird that we're showing the user_id number rather than the username or something like that, especially because the user id won't mean anything to most folks.